### PR TITLE
fix: correct Tweedie deviance formula for power not in {0, 1}

### PR DESCRIPTION
**Pull request by AlejandroCoronadoN**
## Problem

For a general Tweedie `power` (and for the `power == 2` gamma case), the factor
of 2 was only applied to the first deviance term:

```python
2 * ( y^(2-p) / ((1-p)(2-p)) )   # only this term is doubled
- ( y * m^(1-p) / (1-p) )        # not doubled
+ ( m^(2-p) / (2-p) )            # not doubled
```

The unit Tweedie deviance doubles all three terms. Because of this, the result
was wrong for every power other than 0 and 1, and even negative for
`1 < power < 2` (a deviance can never be negative).

```
power=1.5: tweedie_deviance = -6.67   sklearn mean_tweedie_deviance = 0.066
power=2.0: tweedie_deviance =  0.069   sklearn                       = 0.045
```

The bug went unnoticed because the loss comparison test only covered powers 0,
1 and 2, and its own reference for `power == 2` repeated the same mistake.

## Fix

Distribute the `2 *` over all three terms in both the general-power branch and
the `power == 2` branch.

## Verification

After the fix, `tweedie_deviance` matches `sklearn.metrics.mean_tweedie_deviance`
for powers 0, 1, 1.2, 1.5, 1.8, 2.0, 2.5 and 3.0 (max abs diff < 1e-6). The test
reference for `power == 2` is corrected to the same formula, and general-power
cases (1.5 and 2.5) are added to `test_loss`, which fail without the code change
and pass with it. The full `test_losses.py` suite (106 tests, pandas and polars)
passes.

### CODE REVIEW

**Comment by MMenchero** [comment]
Hi @AlejandroCoronadoN thanks for this fix! I just updated the documentation to show the correct formula. 

Can you sign the license so we can merge your PR? We would like to make a release asap, including your fix.

**Comment by nasaul** [comment]
Since user didn't signed the cla I'll close the PR and open one with the correction.
